lib/runtime/wazero/imports.go

```diff
@@ -116,6 +116,88 @@ func ext_crypto_ecdsa_generate_version_1(ctx context.Context, m api.Module, _ ui
 	panic("TODO impl: see https://github.com/ChainSafe/gossamer/issues/3769 ")
 }
 
+type WasmParamType interface {
+	uint32 | uint64 | int32
+}
+
+func wrap0(f func(ctx context.Context, m api.Module)) api.GoModuleFunc {
+	return func(ctx context.Context, m api.Module, stack []uint64) {
+		f(ctx, m)
+	}
+}
+
+func wrap1[T0 WasmParamType](f func(ctx context.Context, m api.Module, a T0)) api.GoModuleFunc {
```

Comment:
You can define a type for the `f` function like to make the functions less verbose and the types clear in the top of the file. 

```go
type noArg func(ctx context.Context, m api.Module)
type singleArg[T WasmParamType] func(ctx context.Context, m api.Module, arg1 T)
...
...
type noArgWithRet[R WasmParamType] func(ctx context.Context, m api.Module) R
...

func wrap0(f noArg) api.GoModuleFunc { ... }

func wrap1[T0 WasmParamType](f singleArg[T]) api.GoModuleFunc { ... }

func wrap0ret[R WasmParamType](f noArgWithRet[R]) api.GoModuleFunc {
```